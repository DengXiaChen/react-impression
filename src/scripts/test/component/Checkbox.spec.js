import React from 'react';
import { shallow } from 'enzyme';
import { Checkbox } from '../../components/impression';

const setup = () => {
    const props = {
        disabled: false,
        defaultChecked: true,
        checked: true,
        onChange: jest.fn(),
        value: 'yes',
    };

    const wrapper = shallow(
        <Checkbox {...props}>
            football
        </Checkbox>
    );

    return {
        props,
        wrapper,
    };
};

describe('Checkbox', () => {
    const { props, wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.checkbox').exists()).toBe(true);
    });

    it('defaultChecked', () => {
        expect(wrapper.find('input').props().checked).toBe(true);
    });

    // it('onChange', () => {
    //     wrapper.find('Checkbox').simulate('click');
    //     expect(wrapper.find('input').props().checked).toBe(false);
    //     expect(props.onChange).toHaveBeenCalled();
    // });

    it('children', () => {
        expect(wrapper.find('.checkbox-label').children().text()).toBe('football');
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
