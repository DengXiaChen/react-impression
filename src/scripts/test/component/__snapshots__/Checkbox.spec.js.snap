// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <label
    className="checkbox"
>
    <input
        checked={true}
        defaultChecked={true}
        disabled={false}
        onChange={[Function]}
        type="checkbox"
    />
    <div
        className="checkbox-addon"
    >
        <i
            className="fa fa-check"
        />
    </div>
    <span
        className="checkbox-label"
    >
        football
    </span>
</label>,
  "nodes": Array [
    <label
      className="checkbox"
>
      <input
            checked={true}
            defaultChecked={true}
            disabled={false}
            onChange={[Function]}
            type="checkbox"
      />
      <div
            className="checkbox-addon"
      >
            <i
                  className="fa fa-check"
            />
      </div>
      <span
            className="checkbox-label"
      >
            football
      </span>
</label>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Checkbox
        checked={true}
        defaultChecked={true}
        disabled={false}
        onChange={[Function]}
        value="yes"
>
        football
</Checkbox>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Checkbox {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "checked": true,
          "children": "football",
          "defaultChecked": true,
          "disabled": false,
          "onChange": [Function],
          "value": "yes",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <label
          className="checkbox"
>
          <input
                    checked={true}
                    defaultChecked={true}
                    disabled={false}
                    onChange={[Function]}
                    type="checkbox"
          />
          <div
                    className="checkbox-addon"
          >
                    <i
                              className="fa fa-check"
                    />
          </div>
          <span
                    className="checkbox-label"
          >
                    football
          </span>
</label>,
        "_debugID": 2,
        "_renderedOutput": <label
          className="checkbox"
>
          <input
                    checked={true}
                    defaultChecked={true}
                    disabled={false}
                    onChange={[Function]}
                    type="checkbox"
          />
          <div
                    className="checkbox-addon"
          >
                    <i
                              className="fa fa-check"
                    />
          </div>
          <span
                    className="checkbox-label"
          >
                    football
          </span>
</label>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Checkbox
    checked={true}
    defaultChecked={true}
    disabled={false}
    onChange={[Function]}
    value="yes"
>
    football
</Checkbox>,
}
`;
